include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -918,6 +995,587 @@ namespace internal
   }
 
 
+  template <int dim, int fe_degree, int n_q_points_1d, typename Number>
+  inline void
+  FEEvaluationImpl<
+    MatrixFreeFunctions::tensor_raviart_thomas,
+    dim,
+    fe_degree,
+    n_q_points_1d,
+    Number>::evaluate(const EvaluationFlags::EvaluationFlags evaluation_flag,
+                      const Number *                         values_dofs_actual,
+                      FEEvaluationData<dim, Number, false> & fe_eval)
+  {
+    if (evaluation_flag == EvaluationFlags::nothing)
+      return;
+
+    // Shape_data is of size 2 with normal direction in [0] and tangential
+    // direction in [1]. See shape_info.template.h
+    std::array<const MatrixFreeFunctions::UnivariateShapeData<Number> *, 2>
+      univariate_shape_data;
+
+    const auto &shape_data = fe_eval.get_shape_info().data;
+
+    univariate_shape_data[0] = &shape_data.front();
+    univariate_shape_data[1] = &shape_data.back();
+
+    Eval_normal eval_normal(univariate_shape_data[0]->shape_values,
+                            univariate_shape_data[0]->shape_gradients,
+                            univariate_shape_data[0]->shape_hessians,
+                            univariate_shape_data[0]->fe_degree + 1,
+                            univariate_shape_data[0]->n_q_points_1d);
```

Comment:
One can create a helper function like `create_evaluator_tensor_product()`

Reply:
Yes, I've done that for the face evaluation, where you have to pick out the right shape functions. (Though I don't think hanging nodes work at this point). But I think doing it here will result in more lines than to just creating these two.